### ISSUE TITLE
Invalid quote appended to autocompletion

### ISSUE THREAD

**Issue by zx8** [fish-shell/fish-shell#6159]
```sh
$ fish --version
fish, version 3.0.2-1534-gcb79d8fa
```

Under the following circumstances, an invalid `"` character is inserted after hitting `<tab>`:

```sh
$ touch foo

$ echo (ls fo ; echo "bar"
             ^
             [cursor]
```

Now, hit `<tab>`:

```sh
$ echo (ls foo" ; echo "bar"
              ^ ^
[invalid quote] [cursor]
```

Ping @krobelus – not sure if this is related to your recent completion changes?

---

**Edit:** Same deal with single quotes.

**Comment by krobelus** [comment]
Works for me on master, I think I actually fixed this in bc533ad939bf7421e27c071c531cc6169c765f19

**Comment by zx8** [comment]
Argh! I was sure I was on the latest `master` commit. I've manually updated to https://github.com/fish-shell/fish-shell/commit/0df464ca2c4113e06fc937c8b0cf7cf045f784d7 and all's well again.

Apologies for the noise.
